components/security-mgt/org.wso2.carbon.security.mgt.ui/src/main/resources/web/keystoremgt/keystore-mgt.jsp

```diff
@@ -154,16 +155,23 @@
                             if (data != null) { //Confusing!!. Sometimes a null object comes. Maybe a bug in Axis!!
                 %>
                 <tr>
-                    <td><%=data.getKeyStoreName()%>
+                    <td><%=Encode.forHtmlContent(data.getKeyStoreName())%>
                     </td>
-                    <td><%=data.getKeyStoreType()%>
+                    <td><%=Encode.forHtmlContent(data.getKeyStoreType())%>
                     </td>
                     <td>
-<a href="import-cert.jsp?keyStore=<%=data.getKeyStoreName()%>" class="icon-link" style="background-image:url(images/import.gif);"><fmt:message key="import.cert"/></a>
-<a href="view-keystore.jsp?keyStore=<%=data.getKeyStoreName()%>" class="icon-link" style="background-image:url(images/view.gif);"><fmt:message key="view"/></a>
-<a href="#" onclick="deleteKeystore('<%=data.getKeyStoreName()%>')" class="icon-link" style="background-image:url(images/delete.gif);"><fmt:message key="delete"/></a>
+<a href="import-cert.jsp?keyStore=<%=Encode.forUriComponent(data.getKeyStoreName())%>"
+   class="icon-link"
+   style="background-image:url(images/import.gif);"><fmt:message key="import.cert"/></a>
+<a href="view-keystore.jsp?keyStore=<%=Encode.forUriComponent(data.getKeyStoreName())%>"
+   class="icon-link"
+   style="background-image:url(images/view.gif);"><fmt:message key="view"/></a>
+<a href="#"
+   onclick="deleteKeystore('<%=Encode.forJavaScriptAttribute(data.getKeyStoreName())%>')" class="icon-link"
+   style="background-image:url(images/delete.gif);"><fmt:message key="delete"/></a>
                     <%if(data.getPubKeyFilePath() != null){ %>
-<a href="<%=data.getPubKeyFilePath()%>" class="icon-link" style="background-image:url(images/view.gif);"><fmt:message key="download.pub.key"/></a>
+<a href="<%=Encode.forUriComponent(data.getPubKeyFilePath())%>" class="icon-link"
```

Comment:
Use Encode.forHtmlAttribute here, since the complete link is untrusted and using URL encoding may break the url